src/go/rdctl/pkg/autostart/autostart_windows.go

```diff
@@ -0,0 +1,84 @@
+package autostart
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/sys/windows/registry"
+)
+
+func EnsureAutostart(autostartDesired bool) error {
+	autostartKey, err := registry.OpenKey(registry.CURRENT_USER, `Software\Microsoft\Windows\CurrentVersion\Run`, registry.SET_VALUE)
+	if err != nil {
+		return fmt.Errorf("failed to open registry key: %w", err)
+	}
+	defer autostartKey.Close()
+
+	if autostartDesired {
+		rancherDesktopPath, err := getRancherDesktopPath()
+		if err != nil {
+			return fmt.Errorf("failed to get path to Rancher Desktop: %w", err)
+		}
+		err = autostartKey.SetStringValue("RancherDesktop", rancherDesktopPath)
+		if err != nil {
+			return fmt.Errorf("failed to set RancherDesktop value: %w", err)
+		}
+	} else {
+		err = autostartKey.DeleteValue("RancherDesktop")
+		if err != nil && !errors.Is(err, registry.ErrNotExist) {
+			return fmt.Errorf("failed to remove RancherDesktop value: %w", err)
```

Comment:
Similarly, maybe "failed to remove Rancher Desktop from auto-start" or some such.